### DISCUSSION TITLE
Update of JCB failing

### DISCUSSION THREAD

**Discussion by outsch** [joomengine/Joomla-Component-Builder#853] [Extention Issues]
I just restarted working on my JCB project after having updated Joomla to 3.10.4 (PHP 8.0.9, Webserver Apache/2.4.38, DB 5.5.68-MariaDB) and found JCB to be updated as well to version: 2.12.14.
However the installation process end with the error message:
`com_componentbuilderInstallerScript::preflight(): Argument #2 ($parent) must be of type Joomla\CMS\Installer\Adapter\ComponentAdapter, Joomla\CMS\Installer\Adapter\PackageAdapter given, called in /var/www/html/boagle/libraries/src/Installer/Adapter/PackageAdapter.php on line 478`
It now seems as if JCB was updated successfully, but not sure whether all is correct given this error. Anyone an idea what the root cause may be?

**Comment by Llewellynvdm** [comment]
JCB is not officially ready for PHP 8

Not sure if that is the issue... but swap out php 8 for php 7.4 and see if the issue continues...

**Comment by outsch** [comment]
OK thanks for the hint, will give it a try
